AWSClientRuntime/Tests/Auth/AWSCredentialProviderTests.swift

```diff
@@ -14,14 +14,46 @@ class AWSCredentialProviderTests: XCTestCase {
     
     func testYouCanUseCustomCredentialsProvider() throws {
         let awsCredsProvider = try AWSCredentialsProvider.fromCustom(MyCustomCredentialsProvider())
-        let credentials = try awsCredsProvider.getCredentials()
+        let credentialsResult = try awsCredsProvider.getCredentials()
+        let credentials = try credentialsResult.get()
+        XCTAssertEqual(credentials.accessKey, "MYACCESSKEY")
+        XCTAssertEqual(credentials.secret, "sekrit")
+    }
+    
+    func testYouCanUseCustomCredentialsProviderFromDiffThread() throws {
+        let awsCredsProvider = try AWSCredentialsProvider.fromCustom(MyCustomThreadTestCredentialsProvider())
+        let credentialsResult = try awsCredsProvider.getCredentials()
+        let credentials = try credentialsResult.get()
         XCTAssertEqual(credentials.accessKey, "MYACCESSKEY")
         XCTAssertEqual(credentials.secret, "sekrit")
     }
 }
 
 struct MyCustomCredentialsProvider: CredentialsProvider {
-    func getCredentials() throws -> AWSCredentials {
-        return AWSCredentials(accessKey: "MYACCESSKEY", secret: "sekrit", expirationTimeout: 30)
+    func getCredentials() throws -> SdkFuture<AWSCredentials> {
+        let future = SdkFuture<AWSCredentials>()
+        future.fulfill(AWSCredentials(accessKey: "MYACCESSKEY", secret: "sekrit", expirationTimeout: 30))
+        return future
+    }
+}
+
+struct MyCustomThreadTestCredentialsProvider: CredentialsProvider {
```

Comment:
Thanks for adding unit tests, but i think these tests should focus on correctness.  We should be able to verify that the call is async in nature, such that it does not block the caller -- if we can write a test that does this too, then great, but if not, i think it's fine to not have a test that does this.

Reply:
k tried to address this in a third unit test I think

Reply:
removing this as not an accurate test